### PR TITLE
EVG-14982: Filtering waterfall charts by status, buildVariant, and task

### PR DESCRIPTION
**Pull request by ziyielenachen**
[EVG-14982](https://jira.mongodb.org/browse/EVG-14982)

### Description 
* Statuses filtering, tasks filtering and buildVariant filtering now reflected on active commit charts
* Removed "tests" from build variant filter
### Screenshots
No filtering:
Screen Shot 2021-07-22 at 1 04 16 PM

Filter by Failures:
Screen Shot 2021-07-22 at 1 04 31 PM

Filter by failures and successes absolute view:
Screen Shot 2021-07-22 at 1 04 53 PM

Filter by failures and success percentage view:
Screen Shot 2021-07-22 at 1 05 12 PM

Filter by dispatched:
Screen Shot 2021-07-22 at 1 05 27 PM

### Evergreen PR 
[https://github.com/evergreen-ci/evergreen/pull/4862](evergreen pr)

### CODE REVIEW

**Review by arjunpatel100 — APPROVED** [review]

File: src/gql/queries/get-mainline-commits.graphql [hunk 1]
```diff
@@ -1,4 +1,7 @@
-query MainlineCommits($options: MainlineCommitsOptions!) {
+query MainlineCommits(
+  $options: MainlineCommitsOptions!
+  $taskStatusCountsOptions: BuildVariantOptions!
```
Comment (khelif96) [inline src/gql/queries/get-mainline-commits.graphql:3]: can we name this `$buildVariantOptions` Since we will reuse it for the buildVariant filters

File: src/pages/Commits.tsx [hunk 2]
```diff
@@ -51,21 +73,26 @@ export const Commits = () => {
     }
   }, [chartTypeParam, setCurrentChartType]);
 
-  usePageTitle(`Project Health | ${projectId}`);
+  // query mainlineCommits data
+  const options = { projectID: projectId, limit: 5 };
+  const taskStatusCountsOptions = {
```
Comment (khelif96) [inline src/pages/Commits.tsx:78]: buildVariantOptions

File: src/pages/commits/CommitsWrapper.tsx [hunk 3]
```diff
@@ -78,12 +78,14 @@ const StyledSkeleton = styled(Skeleton)`
   margin-top: 12px;
 `;
 
-export const FlexRowContainer = styled.div`
+// If we have more than four commits, container should expand entire width
+// Else they should align left
+export const FlexRowContainer = styled.div<{ numCommits: number }>`
+  width: ${({ numCommits }) => (numCommits > 4 ? 1 : numCommits / 7) * 100}%;
```
Comment (khelif96) [inline src/pages/commits/CommitsWrapper.tsx:84]: Why is this necessary?
